src/PostgREST/QueryBuilder.hs

```diff
@@ -61,9 +61,22 @@ import           Data.Scientific         ( FPFormat (..)
 import           Prelude hiding          (unwords)
 import           PostgREST.ApiRequest    (PreferRepresentation (..))
 
+{-| The generic query result format used by API responses. The location header
+    is represented as a list of strings containing variable bindings like
+    @"k1=eq.42"@. If unused, it's null/Nothing rather than the empty list
+    because 'PostgreSQL.Binary.Decoder.arrayDimension' cannot decode an empty
+    array!
```

Comment:
Very helpful comment.

Do you know if this is a bug in `arrayDimension`? It looks like an intentional library decision: https://github.com/nikita-volkov/postgresql-binary/blob/da82acf62855012ddecdcd6378d8fba69ca77b31/library/PostgreSQL/Binary/Decoder.hs#L499-L502

@nikita-volkov can you give us more context about this function?